transforms/universal/html2parquet/python/src/html2parquet_transform.py

```diff
@@ -0,0 +1,84 @@
+import time
+from argparse import ArgumentParser, Namespace
+from typing import Any
+import zipfile
+import io
+import trafilatura
+from datetime import datetime
+
+
+import pyarrow as pa
+
+# disabled for now
+# from data_processing_ray.runtime.ray import RayTransformLauncher
+# from data_processing_ray.runtime.ray.runtime_configuration import (
+#   RayTransformRuntimeConfiguration,
+# )
+# import data_processing
+
+
+from data_processing.transform import AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, get_logger, TransformUtils
+
+
+class HtmlToParquetTransform(AbstractBinaryTransform):
+    def __init__(self, config: dict[str, Any]):
+        pass
+
+    def transform_binary(self, file_name: str, byte_array: bytes) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Converts raw data file (ZIP) to Parquet format
+        """
+        # We currently only process .zip files
+        if TransformUtils.get_file_extension(file_name)[1] != ".zip":
+            self.logger.warning(f"Got unsupported file type {file_name}, skipping")
+            return [], {}
+        data = []
+        number_of_rows = 0
+
+        with zipfile.ZipFile(io.BytesIO(bytes(byte_array))) as opened_zip:
+            # Loop through each file member in the ZIP archive
+            for member in opened_zip.infolist():
+                if not member.is_dir() and '__MACOSX' not in member.filename:
+                    with opened_zip.open(member) as file:
+                        try:
+                            # Read the content of the file
+                            content_bytes = file.read()
+                            # Use Trafilatura library
+                            content_string = trafilatura.extract(content_bytes)
```

Comment:
You need to handle None return value, which is apparently an allowed/expected return.